### PR TITLE
Port fix for IDNA test

### PR DESCRIPTION
**Pull request by danmosemsft**
Fix https://github.com/dotnet/corefx/issues/23928

Port commit 8dc08be6eb99ce6475f087b4d7418f3e477e55cb
Author: Hugh Bellamy <hughbellars@gmail.com>
Date:   Sat Jun 24 00:12:36 2017 +0700

    Make IdnaConformance tests easier to diagnose when they go wrong (#21463)

Port commit 70790f8d9760ff1060f2f853e4427feb21a0d08c
Author: Hugh Bellamy <hughbellars@gmail.com>
Date:   Sat Jun 24 09:19:28 2017 +0700

    Fix failing Globalization test for Windows updates and update GetCodePoints (#21488)

### CODE REVIEW

**Comment by danmosemsft** [comment]
@dotnet-bot help please

**Comment by danmosemsft** [comment]
https://github.com/dotnet/corefx/issues/23928

File: src/System.Globalization.Extensions/tests/IdnMapping/IdnMappingIdnaConformanceTests.cs [hunk 1]
```diff
@@ -41,25 +49,27 @@ public void GetAscii_Success()
...
             {
-                if (entry.UnicodeResult.Success && entry.UnicodeResult.ValidDomainName)
+                try
                 {
-                    try
-                    {
-                        var map = new IdnMapping { UseStd3AsciiRules = true, AllowUnassigned = true };
-                        var unicodeResult = map.GetUnicode(entry.Source);
+                    var map = new IdnMapping { UseStd3AsciiRules = true, AllowUnassigned = true };
+                    var unicodeResult = map.GetUnicode(entry.Source);
```
Comment (stephentoub) [inline src/System.Globalization.Extensions/tests/IdnMapping/IdnMappingIdnaConformanceTests.cs]: Nit: var => ?

File: src/System.Globalization.Extensions/tests/IdnMapping/IdnMappingIdnaConformanceTests.cs [hunk 2]
```diff
@@ -41,25 +49,27 @@ public void GetAscii_Success()
...
 
-                        Assert.Equal(entry.UnicodeResult.Value, unicodeResult, StringComparer.OrdinalIgnoreCase);
-                    }
-                    catch (ArgumentException)
+                    Assert.Equal(entry.UnicodeResult.Value, unicodeResult, StringComparer.OrdinalIgnoreCase);
+                }
+                catch (ArgumentException)
+                {
+                    if (!string.Equals(entry.UnicodeResult.Value, entry.Source, StringComparison.OrdinalIgnoreCase))
```
Comment (stephentoub) [inline src/System.Globalization.Extensions/tests/IdnMapping/IdnMappingIdnaConformanceTests.cs]: If they are equal we just ignore the error?

Comment (danmosemsft) [reply src/System.Globalization.Extensions/tests/IdnMapping/IdnMappingIdnaConformanceTests.cs]: I'll look in master. Here I will port as is.

File: src/System.Globalization.Extensions/tests/IdnMapping/IdnMappingIdnaConformanceTests.cs (hunk 1 above)
Comment (danmosemsft) [reply src/System.Globalization.Extensions/tests/IdnMapping/IdnMappingIdnaConformanceTests.cs]: Will fixin master

**Comment by stephentoub** [comment]
@danmosemsft, are you opening PRs for the other release branches, too?

**Comment by tarekgh** [comment]
Thanks, @danmosemsft for fixing this.

Just FYI, the original change caused this failure is coming from Windows. I have fixed these in many branches (master, PN and corert...etc). these were not failing on 1.x and 2.0.x before as we were running on Windows version which didn't have the change. but it looks the CI machines get upgrading to new version.

**Comment by danmosemsft** [comment]
This should be fixed in 1.1 and 1.0 by 545580a e86a00f on Friday.

**Comment by stephentoub** [comment]
The fix is also needed in the release/uwp6.0 branch.

**Comment by stephentoub** [comment]
> This should be fixed in [...] 1.0

Doesn't appear to be:
https://github.com/dotnet/corefx/pull/23818#issuecomment-328437386

**Comment by tarekgh** [comment]
> The fix is also needed in the release/uwp6.0 branch.

I am seeing the fix is there 
https://github.com/dotnet/corefx/blob/release/uwp6.0/src/System.Globalization.Extensions/tests/IdnMapping/Data/Unicode_6_0/IdnaTest_6.txt#L769

**Comment by stephentoub** [comment]
> I am seeing the fix is there

Ah, you're right, PRs are failing to release/uwp6.0 due to the stack trace issue rather than to this:
https://ci3.dot.net/job/dotnet_corefx/job/release_uwp6.0/job/windows-TGroup_netfx+CGroup_Release+AGroup_x86+TestOuter_false_prtest/176/consoleText

**Pull request merged**